### PR TITLE
[REEF-881] Create a ProgressProvider on the C# side

### PR DESCRIPTION
**Pull request by afchung**
This addressed the issue by
  * Creating the ProgressProvider class in C# and piping it to Java.

JIRA:
  [REEF-881](https://issues.apache.org/jira/browse/REEF-881)

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.Bridge/JavaClrBridge.cpp [hunk 1]
```diff
@@ -563,6 +565,24 @@ JNIEXPORT void JNICALL Java_org_apache_reef_javabridge_NativeInterop_clrSystemDr
 	}
 }
 
+/*
+* Class:     org_apache_reef_javabridge_NativeInterop
```
Comment (dongjoon-hyun) [inline lang/cs/Org.Apache.REEF.Bridge/JavaClrBridge.cpp]: Please add a space.

File: lang/cs/Org.Apache.REEF.Bridge/JavaClrBridge.cpp [hunk 2]
```diff
@@ -621,6 +641,9 @@ static JNINativeMethod methods[] = {
 
 	{ "clrSystemDriverRestartCompletedHandlerOnNext", "(JLorg/apache/reef/javabridge/generic/DriverRestartCompletedBridge;)V",
 	(void*)&Java_org_apache_reef_javabridge_NativeInterop_clrSystemDriverRestartCompletedHandlerOnNext },
+
+    { "clrSystemProgressProviderGetProgress", "(J)F",
```
Comment (dongjoon-hyun) [inline lang/cs/Org.Apache.REEF.Bridge/JavaClrBridge.cpp]: Please add a tab here. This file seems to have a mixed indentation style.

File: lang/cs/Org.Apache.REEF.Driver/Defaults/DefaultProgressProvider.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
+﻿/**
+* Licensed to the Apache Software Foundation (ASF) under one
```
Comment (dongjoon-hyun) [inline lang/cs/Org.Apache.REEF.Driver/Defaults/DefaultProgressProvider.cs]: Please add a space.

**Comment by dongjoon-hyun** [comment]
I just leaved a few comments about indentation.

**Comment by dongjoon-hyun** [comment]
@afchung , it looks good to me. I'll merge this.

**Comment by afchung** [comment]
Thanks @dongjoon-hyun!

**Comment by dongjoon-hyun** [comment]
:) @afchung

**Pull request closed without merging**